test/example/type-builder.cpp

```diff
@@ -0,0 +1,137 @@
+#include <cassert>
+#include <iostream>
+
+#include "wasm-type.h"
+
+using namespace wasm;
+
+// Construct Signature, Struct, and Array heap types using undefined types.
+void test_builder() {
+  std::cout << ";; Test TypeBuilder\n";
+
+  // (type $sig (func (param (ref $struct)) (result (ref $array) i32)))
+  // (type $struct (struct (field (ref null $array) (mut ref extern))))
+  // (type $array (array (mut ref null extern)))
+
+  TypeBuilder builder(3);
+
+  Type refSig = builder.getTempRefType(0, false);
+  Type refStruct = builder.getTempRefType(1, false);
+  Type refArray = builder.getTempRefType(2, false);
+  Type refNullArray = builder.getTempRefType(2, true);
+  Type refExt(HeapType::ext, false);
```

Comment:
Unfortunately we can't use `extern` because it is a keyword. Between the alternatives `extern_` and `ext` I thought `ext` looked better, but perhaps the ugliness of `extern` would be worth the clarity. @kripken, what do you think about this? I'd be happy to change it if we collectively think `extern_` would be better.

Reply:
I have a slight preference to `extern_`.

Reply:
Sounds good. I'll make this change in a followup 👍